### ISSUE TITLE
286 compatible dos version request

### ISSUE THREAD

**Issue by p1plp1** [freeserf/freeserf#459]
original game runs in protected mode and it is also inpossible to run it on 8086 8088 80286 etc. even with 16 or 32mb ram, because 8086 and 8088 doesnt support protected mode, 80286 suppports protected mode but in different way - it is impossible to leave from this special protected mode and it differs from 386 protected mode ... i think it is possible to compile only with 286 instruction set and little bit optimize, if it is possible to run it on amiga ;)

**Comment by sternenzaehler** [comment]
thats a good one:)
what is the operating system there ? plain old DOS?

**Comment by sternenzaehler** [comment]
I looked up Amiga and it has a 32 bit processor, actually, the Motorola 68K, which is 16/32 bit, the registers and the instruction set are 32 bit.
80286 has no 32 bit registers. though in theory of course you do not need them, enough to use 16 bit registers.

**Comment by p1plp1** [comment]
plain dos mode, 8087 or 287 fpu support shoud be nice, im thinking about it, 8086 and 8088 also didnt support some instructions, so for example push all and pop all instructions are unsupported, only V20 cpu supports such things from 186+ instruction set.

**Comment by p1plp1** [comment]
286 is able to leave protected mode with a little help - reset from keyboard controller :D

**Comment by sternenzaehler** [comment]
in theory I could remember my childhood and try to port the code to OpenWatcom, which is to my mind the best chance to make it work on a 286. Would be interesting, but is kinda scary as it may become a very difficult task (if it is bad luck) :)
I will try to look into it and make an assessment.
I do think the most problems will arise from sound, 286 means good old soundblaster and ports and IRQs :)

**Comment by p1plp1** [comment]
yes, watcom compiler, and about sound on 286 or pc/xt, soudcards able to play on such computers:
-SB16 compatible
-some SB PRO cards
-SB CMS
-cards with yamaha YM3812
-yamaha with OPL2, 3, 4 (interesting is that YMF71xx 16bit isa cards works)
-adlib
-gravis ultrasound (any version)
i dont have ADLIB yet, others im able to test :)

**Comment by p1plp1** [comment]
hmm im allready thinking about it, objects shoud be replaced with structures, procedures and functions, looks like it maybe can be easier than fixing that object refresh problem in freeserf loop :D

**Comment by p1plp1** [comment]
vga cards i have and im able to test:
OTIvga (some models 8bit 16 bit isa)
trident microsystems (8800 8900 9000 many models)
wdc graphics (some models)
chips and technologies (some models)
